### PR TITLE
cleanup: fix race condition

### PR DESCRIPTION
**Pull request by BinacsLee**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

First part of the work for https://github.com/kubernetes-sigs/kueue/issues/67

Second part: https://github.com/kubernetes/test-infra/pull/25443

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes-sigs/kueue/issues/67

#### Special notes for your reviewer:

None

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -41,6 +41,7 @@ endif
 
 GO_CMD ?= go
 GO_FMT ?= gofmt
+GO_TEST_FLAGS ?= -race -count 1
```
Comment (alculquicondor) [inline Makefile:44]: just `-race`, as `-count 1` is the default.

File: pkg/queue/manager_test.go [hunk 2]
```diff
@@ -478,9 +478,7 @@ func TestHeadsAsync(t *testing.T) {
 		if err := manager.AddQueue(ctx, &q); err != nil {
 			t.Errorf("Failed adding queue: %s", err)
 		}
-		go func() {
-			manager.addOrUpdateWorkload(&wl)
```
Comment (alculquicondor) [inline pkg/queue/manager_test.go:482]: This was intentional, but it should have been `manager.AddOrUpdateWorkload`, which takes the lock.

File: pkg/scheduler/scheduler.go [hunk 3]
```diff
@@ -217,7 +217,7 @@ func (s *Scheduler) assign(ctx context.Context, e *entry) error {
 	go func() {
 		err := s.client.Update(ctx, newWorkload)
 		if err == nil {
-			s.recorder.Eventf(newWorkload, corev1.EventTypeNormal, "Assigned", fmt.Sprintf("Assigned to capacity %v", e.Capacity))
+			s.recorder.Eventf(newWorkload, corev1.EventTypeNormal, "Assigned", fmt.Sprintf("Assigned to capacity %v", newWorkload.Spec.AssignedCapacity))
```
Comment (alculquicondor) [inline pkg/scheduler/scheduler.go:220]:
```suggestion
			s.recorder.Eventf(newWorkload, corev1.EventTypeNormal, "Assigned", "Assigned to capacity %v", newWorkload.Spec.AssignedCapacity)
```

Eventf already knows how to do formatting.

**Comment by BinacsLee** [comment]
hi @alculquicondor , thanks for review! please take another look ＾ ＾

File: pkg/queue/manager_test.go (hunk 2 above)
Comment (alculquicondor) [reply pkg/queue/manager_test.go:482]: Leave the `go` call please

Comment (BinacsLee) [reply pkg/queue/manager_test.go:482]: done

**Comment by alculquicondor** [comment]
@BinacsLee it seems that we have another race in a log line somewhere...
could you take a look?

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kueue/93/pull-kueue-test-unit-main/1499496533626720256

**Comment by BinacsLee** [comment]
> @BinacsLee it seems that we have another race in a log line somewhere... could you take a look?
> 
> https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kueue/93/pull-kueue-test-unit-main/1499496533626720256

sure, I'll deal with it as soon as possible

**Pull request merged**